packages/jsii/lib/assembler.ts

```diff
@@ -1788,6 +1791,60 @@ export class Assembler implements Emitter {
     return Promise.resolve(jsiiType);
   }
 
+  private assertNoDuplicateEnumValues(decl: ts.EnumDeclaration): void {
+    type EnumValue = {
+      name: string;
+      value: string;
+      decl: ts.DeclarationName | undefined;
+    };
+
+    const enumValues = decl.members
+      .filter((m) => m.initializer)
+      .map((member): EnumValue => {
+        return {
+          value: member.initializer!.getText(),
+          name: member.name.getText(),
+          decl: ts.getNameOfDeclaration(member),
+        };
+      });
+
+    const hasDuplicateEnumValues = enumValues.some(
+      (val, _, arr) => arr.filter((e) => val.value === e.value).length > 1,
+    );
+
+    if (hasDuplicateEnumValues) {
+      const enumValueMap = enumValues.reduce<Record<string, EnumValue[]>>(
+        (acc, val) => {
+          if (!acc[val.value]) {
+            acc[val.value] = [];
+          }
+          acc[val.value].push(val);
+          return acc;
+        },
+        {},
+      );
+      for (const duplicateValue of Object.keys(enumValueMap)) {
+        if (enumValueMap[duplicateValue].length > 1) {
+          const err = JsiiDiagnostic.JSII_1004_DUPLICATE_ENUM_VALUE.create(
+            enumValueMap[duplicateValue][0].decl!,
+            `Value ${duplicateValue} is used for multiple enum values: ${enumValueMap[
```

Comment:
You'll want to move the template string up to the `JsiiDiagnostic` entry. The `formatter` function is the mechanism by which we aim to guarantee that the error message is always consistent for a given error class (in the future, that same error could be emitted from another place, and then we'd need to make sure the error messages are in synch).

Reply:
✅